frontend/src/lib/k8s/apiProxy.ts

```diff
@@ -35,6 +37,64 @@ export interface ClusterRequest {
   certificateAuthorityData?: string;
 }
 
+//refreshToken checks if the token is about to expire and refreshes it if so.
+async function refreshToken(token: string | null) {
+  if (!token) {
+    return;
+  }
+  // decode token
+  const decodedToken: any = decodeToken(token);
+
+  // return if the token doesn't have an expiry time
+  if (!decodedToken.exp) {
+    return;
+  }
+  // convert expiry seconds to date object
+  const expiry = decodedToken.exp;
+  const now = new Date().valueOf();
+  const expDate = new Date(0);
+  expDate.setUTCSeconds(expiry);
+
+  // calculate time to expiry in minutes
+  const diff = (expDate.valueOf() - now) / 60000;
+  // If the token is about to expire, refresh it
+  if (diff < 10) {
```

Comment:
Also, I think early returning if the token is young enough may make the code cleaner.